client/packages/invoices/src/Prescriptions/LineEditView/LineEditView.tsx

```diff
@@ -77,81 +95,62 @@ export const PrescriptionLineEditView = () => {
   useEffect(() => {
     setCustomBreadcrumbs({
       1: data?.invoiceNumber.toString() ?? '',
-      2: currentItem?.name || '',
+      2: item?.name || '',
     });
-  }, [currentItem, data?.invoiceNumber, itemId]);
+  }, [item, data?.invoiceNumber, itemId]);
 
   useConfirmOnLeaving(
     'prescription-line-edit',
     // Need a custom checking method here, as we don't want to warn user when
     // switching to a different item within this page
     {
       customCheck: {
-        navigate: (current, next) => {
-          if (!isDirty.current) return false;
-
-          const currentPathParts = current.pathname.split('/');
-          const nextPathParts = next.pathname.split('/');
-          // Compare URLS, but don't include the last part, which is the ItemID
-          currentPathParts.pop();
-          nextPathParts.pop();
-          return !isEqual(currentPathParts, nextPathParts);
-        },
+        navigate: () => isDirty.current,
         refresh: () => isDirty.current,
       },
     }
   );
 
-  const updateAllLines = (lines: DraftPrescriptionLine[]) => {
-    if (itemId === 'new') {
-      newItemId.current = lines[0]?.item.id;
-    }
-
-    if (typeof itemId === 'string')
-      setAllDraftLines({ ...allDraftLines, [itemId]: lines });
-  };
+  // TODO: We need a better solution for this long term!
+  useEffect(() => {
+    // Using useAllocationContext isDirty for rest of form, need to sync it with
+    // the isDirty state for confirm on leaving
+    isDirty.current = allocationIsDirty;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [allocationIsDirty]);
 
   const onSave = async () => {
-    if (!isDirty.current) return;
-
-    const flattenedLines = Object.values(allDraftLines).flat();
-
-    // needed since placeholders aren't being created for prescriptions yet, but
-    // still adding to array
-    const isOnHold = flattenedLines.some(
-      ({ stockLine, location }) => stockLine?.onHold || location?.onHold
-    );
-
-    const patch =
-      status !== InvoiceNodeStatus.Picked &&
-      flattenedLines.length >= 1 &&
-      !isOnHold
-        ? {
-            id: invoiceId,
-            status: InvoiceNodeStatus.Picked,
-          }
-        : undefined;
-
-    await saveLines({
-      draftPrescriptionLines: flattenedLines,
-      patch,
-    });
+    const contextItemId = item?.id ?? itemId ?? newItemId.current;
+    if (!contextItemId) {
+      alert('No itemId found');
```

Comment:
Should probably not be an alert...

Reply:
[2c44325](https://github.com/msupply-foundation/open-msupply/pull/7888/commits/2c44325d039c71ebba43634a2843a1bdb472c048)